crates/sui-move/src/lib.rs

```diff
@@ -45,9 +45,13 @@ pub struct Calib {
 
 pub fn execute_move_command(
     package_path: Option<PathBuf>,
-    build_config: BuildConfig,
+    mut build_config: BuildConfig,
     command: Command,
 ) -> anyhow::Result<()> {
+    if build_config.default_flavor.is_none() {
+        // set the compilation flavor to Sui unless user already set it to another flavor
```

Comment:
I think we should assert it is Sui if we are using the `sui-move` tool chain. I think I have that check somewhere...

Reply:
It is not set to anything at this point unless the user set it. In the other PR where I was trying to fix the double-compilation problem, I simply set it here to `Flavor::Sui` but that silently overrides a flavor set by the user. Should we then:
- set it it to `Flavor::Sui`
- report and error if it's set to something else than `Flavor::Sui` at this point in the code as we are in the Sui toolchain here
?

Reply:
We do it for the package resolver (which I think is used by most things?)
https://github.com/MystenLabs/sui/blob/main/crates/sui-move-build/src/lib.rs#L201